### ISSUE TITLE
unregistering singleton.

### ISSUE THREAD

**Issue by matejthetree** [fluttercommunity/get_it#3]
Do you mind implementing simple unregister function, If I dont need the service anymore?

**Comment by matejthetree** [comment]
I can do it also, if you agree that it is a good idea

**Comment by escamoteur** [comment]
Not sure when you would need that as you can reset the whole GetIt but if you make a PR, sure

**Comment by ejabu** [comment]
> Not sure when you would need that as you can reset the whole GetIt but if you make a PR, sure

I dont understand how to reset the whole GetIt. Could you confirm one of these ?

```
GetIt sl = new GetIt();

void setup() {
  sl.allowReassignment = true;
  // Register API Client
  sl.registerSingleton<WeatherAPI>(new WeatherAPIOpenWeatherMap() );
  sl.registerSingleton<AppModel>(new AppModel());
}
```

and then changed by this ?

```
MaterialButton(
            child: Text("Toggle"),
            onPressed: (){

              // Changed to Mock
              sl.registerSingleton<WeatherAPI>(new WeatherAPIEmulation() );
            },
          )


```

best regards

**Comment by escamoteur** [comment]
Hi,
sorry I missed your ticket.

GetIt has a `reset()` method to reset it completely.

You can do whar you do above if you set `allowReassignment=true`

**Comment by escamoteur** [comment]
This is available with version V2.1.0
